aicsimageio/readers/ome_zarr_reader.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from typing import Any, Dict, List, Optional, Tuple
+
+import xarray as xr
+from fsspec.spec import AbstractFileSystem
+
+from .. import constants, exceptions, types
+from ..dimensions import DimensionNames
+from ..metadata import utils as metadata_utils
+from ..utils import io_utils
+from .reader import Reader
+
+try:
+    from ome_zarr.io import parse_url
+    from ome_zarr.reader import Reader as ZarrReader
+
+except ImportError:
+    raise ImportError(
+        "ome_zarr is required for this reader. " "Install with `pip install 'ome_zarr'`"
+    )
+
+###############################################################################
+
+
+class OmeZarrReader(Reader):
+    """
+    Wraps the ome-zarr-py API to provide the same aicsimageio Reader API but for
+    OmeZarr images.
+
+    Parameters
+    ----------
+    image: types.PathLike
+        Path to image file to construct Reader for.
+    fs_kwargs: Dict[str, Any]
+        Any specific keyword arguments to pass down to the fsspec created filesystem.
+        Default: {}
+    """
+
+    @staticmethod
+    def _is_supported_image(fs: AbstractFileSystem, path: str, **kwargs: Any) -> bool:
+        try:
+            ZarrReader(parse_url(path, mode="r"))
+            return True
+
+        except AttributeError:
+            return False
+
+    def __init__(
+        self,
+        image: types.PathLike,
+        fs_kwargs: Dict[str, Any] = {},
+    ):
+        # Expand details of provided image
+        self._fs, self._path = io_utils.pathlike_to_fs(
+            image,
+            enforce_exists=False,
+            fs_kwargs=fs_kwargs,
+        )
+
+        # Enforce valid image
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+        self._zarr = ZarrReader(parse_url(self._path, mode="r")).zarr
```

Comment:
There may be problems with this... I think this currently only works if the image is public read? The `_path` is decoupled from the `_fs` itself and if the user needs to provide credentials, then this won't work.

UNLESS, the user has the credentials in the environment. It might be good to note that the ZarrReader needs env creds and not provided creds.

Reply:
I agree with documenting this and maybe catching such a failure if possible, but I would say it's ok if our first zarr reader is "public data or supply your own creds in env".    Args creds could be a second revision...